config/action.d/freebsd-ipfw.conf

```diff
@@ -0,0 +1,99 @@
+# Fail2Ban configuration file
+#
+# This file was created to provide ipfw with a configuration for FreeBSD
+# as flexible as possible
+# Author:	Nilton J. Rizzo
+#
+#
+
+[Definition]
+
+# Option:  actionstart
+# Notes.:  command executed once at the start of Fail2Ban.
+# Values:  CMD
+#
+actionstart = 
+
+
+# Option:  actionstop
+# Notes.:  command executed once at the end of Fail2Ban
+# Values:  CMD
+#
+actionstop =  ipfw delete <rulenumber>
+
+
+# Option:  actioncheck
+# Notes.:  command executed once before each actionban command
+# Values:  CMD
+#
+actioncheck = 
+
+
+# Option:  actionban
+# Notes.:  command executed when banning an IP. Take care that the
+#          command is executed with Fail2Ban user rights.
+# Tags:    See jail.conf(5) man page
+# Values:  CMD
+#
+#actionban = ipfw add <rulenumber> <blocktype> <proto> from <ip> to <localhost> dst-port <port> in via <nic> 
+actionban = ipfw add <rulenumber> <blocktype> <proto> from <ip> to <localhost> <port>
+
```

Comment:
You did not do any check if there were any pre-existing ipfw rules at this number.
As a result, when you stop or reload fail2ban, your `actionstop` will delete the pre-existing ipfw rule.

You can check how this is implemented in the existing action.d/bsd-ipfw.conf
While the script in there might look a bit convoluted, it works well. In your case, it will require a very complicated logic hook to verify that there were no pre-existing firewall rules, because you are creating new rules and not setting any flags that would allow distinguishing whose rule(s) is sitting at this number.